### PR TITLE
[BREAKING] Remove --old-reporter from solc

### LINKED ISSUE #7369: [CLI] Remove old error reporter
https://github.com/ethereum/solidity/pull/7368 enables the new error reporter, but keeps the old one as command line option. It should be removed with the next breaking release.

### PR DESCRIPTION
**Pull request by axic**
Closes #7369.

The question from #7369 remains: StandardCompiler uses the old reporter for the `formattedMessage` field. Do we want to change that too?

### CODE REVIEW

**Comment by leonardoalt** [comment]
I think it would make sense to remove from both.

**Comment by chriseth** [comment]
If we want to get rid of it, we should remove it everywhere.

**Comment by leonardoalt** [comment]
@axic ping
seems to be consensus to remove it from JSON as well

**Comment by axic** [comment]
> seems to be consensus to remove it from JSON as well

I know, we discussed it yesterday. It is in progress, but it is quite a bit of work, especially fixing all the tests.

**Comment by axic** [comment]
Reminder: this is waiting for a review. It makes no sense (for me) to keep updating the test expectations as many other PRs are touching those. I'll of course do a final test update after there's some agreement of the review and we are close to merging.

**Comment by leonardoalt** [comment]
@axic so the code is ready?

**Comment by axic** [comment]
The code is. Tests needs to be updated once-again before merge, and need to fix solc-js too.

File: test/libsolidity/InlineAssembly.cpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ std::optional<Error> parseAndReturnFirstError(
 		{
 			string errors;
 			for (auto const& err: stack.errors())
-				errors += SourceReferenceFormatter::formatErrorInformation(*err);
+				errors += SourceReferenceFormatterHuman::formatErrorInformation(*err);
```
Comment (leonardoalt) [inline test/libsolidity/InlineAssembly.cpp]: Why do the others also use `, true, false` in this doesn't?

File: test/libsolidity/AnalysisFramework.cpp [hunk 2]
```diff
@@ -149,7 +149,7 @@ string AnalysisFramework::formatErrors() const
 
 string AnalysisFramework::formatError(Error const& _error) const
 {
-	return SourceReferenceFormatter::formatErrorInformation(_error);
+	return SourceReferenceFormatterHuman::formatErrorInformation(_error);
```
Comment (leonardoalt) [inline test/libsolidity/AnalysisFramework.cpp]: Also this

File: libsolidity/codegen/ir/IRGenerator.cpp [hunk 3]
```diff
@@ -60,7 +60,7 @@ pair<string, string> IRGenerator::run(
 	{
 		string errorMessage;
 		for (auto const& error: asmStack.errors())
-			errorMessage += langutil::SourceReferenceFormatter::formatErrorInformation(*error);
+			errorMessage += langutil::SourceReferenceFormatterHuman::formatErrorInformation(*error);
```
Comment (leonardoalt) [inline libsolidity/codegen/ir/IRGenerator.cpp]: Why not `withErrorIds`?

File: libsolidity/codegen/CompilerContext.cpp [hunk 4]
```diff
@@ -460,7 +460,7 @@ void CompilerContext::appendInlineAssembly(
 			_assembly + "\n"
 			"------------------ Errors: ----------------\n";
 		for (auto const& error: errorReporter.errors())
-			message += SourceReferenceFormatter::formatErrorInformation(*error);
+			message += SourceReferenceFormatterHuman::formatErrorInformation(*error);
```
Comment (leonardoalt) [inline libsolidity/codegen/CompilerContext.cpp]: Also here (withErrorids)

File: test/libsolidity/InlineAssembly.cpp (hunk 1 above)
Comment (axic) [reply test/libsolidity/InlineAssembly.cpp]: This is for testing and the defaults are fine. I used the explicit settings to turn off colour where desired.

Comment (leonardoalt) [reply test/libsolidity/InlineAssembly.cpp]: Ok

File: libsolidity/codegen/ir/IRGenerator.cpp (hunk 3 above)
Comment (axic) [reply libsolidity/codegen/ir/IRGenerator.cpp]: Not sure, is that useful? If it is we can add it.

Comment (leonardoalt) [reply libsolidity/codegen/ir/IRGenerator.cpp]: I didn't mean to change the current behavior, just wondering if this is the same as the current behavior

**Comment by axic** [comment]
@bshastry need feedback on the fuzzer changes.

**Comment by bshastry** [comment]
> @bshastry need feedback on the fuzzer changes.

LGTM!

File: libsolidity/codegen/CompilerContext.cpp [hunk 5]
```diff
@@ -460,7 +460,7 @@ void CompilerContext::appendInlineAssembly(
 			_assembly + "\n"
 			"------------------ Errors: ----------------\n";
 		for (auto const& error: errorReporter.errors())
-			message += SourceReferenceFormatter::formatErrorInformation(*error);
```
Comment (chriseth) [inline libsolidity/codegen/CompilerContext.cpp]: Why not rename it?

**Comment by chriseth** [comment]
Test failures and needs rebase.

**Comment by axic** [comment]
> Test failures 

All tests were updated 2 rebases ago, but then I removed it because no point updating it (painfully), unless I get feedback on the actual code changes.

File: libsolidity/codegen/CompilerContext.cpp (hunk 5 above)
Comment (axic) [reply libsolidity/codegen/CompilerContext.cpp]: Could do.

**Review by leonardoalt — APPROVED** [review]
Code looks good, but needs to be rebased&merged quickly to avoid having to fix newly merged changes.

**Comment by chriseth** [comment]
I would propose to fix the solcjs failure after this has been merged.

**Comment by axic** [comment]
> I would propose to fix the solcjs failure after this has been merged.

Do we have a breaking branch for solcjs? Do we want one? Should we just loosen up the check in the test to search for `SyntaxError: Invalid pragma "contract"` ?

I think we are good enough with this: https://github.com/ethereum/solc-js/pull/504

**Pull request merged**